### PR TITLE
add feature flag to block the api request

### PR DESCRIPTION
**Pull request by acosferreira**
# Description
Enable use of feature flag to hide the option to create edge-groups and avoid data leak during the migration execution

Fixes # (THEEDGE-3576)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

File: unleash/features/feature.go [hunk 1]
```diff
@@ -120,6 +120,9 @@ var StaticDeltaShortCircuit = &Flag{Name: "edge-management.static_delta_shortcir
 // StaticDeltaGenerate toggles creation of static deltas
 var StaticDeltaGenerate = &Flag{Name: "edge-management.static_delta_generate", EnvVar: "FEATURE_STATIC_DELTA_GENERATE"}
 
+// CreateGroup toggles creation of static deltas
+var CreateGroup = &Flag{Name: "edgeParity.create-group", EnvVar: "FEATURE_EDGE_PARITY_CREATE_GROUP"}
```
Comment (ldjebran) [inline unleash/features/feature.go:124]:
```
var EdgeParityCreateGroup   = &Flag{Name: "edgeParity.groups-create-group", EnvVar: "FEATURE_EDGE_PARITY_CREATE_GROUP"}
```
I really prefer the root "edgeParity.groups-" for all feature flags related to groups, what do you think ?

File: pkg/routes/devicegroups.go [hunk 2]
```diff
@@ -319,29 +320,35 @@ func GetAllDeviceGroups(w http.ResponseWriter, r *http.Request) {
...
+			respondWithAPIError(w, ctxServices.Log, apiError)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		respondWithJSONBody(w, ctxServices.Log, &deviceGroup)
+	} else {
+		w.WriteHeader(http.StatusUnauthorized)
+		respondWithJSONBody(w, ctxServices.Log, feature.CreateGroup.IsEnabled())
+	}
```
Comment (ldjebran) [inline pkg/routes/devicegroups.go:351]: I think it's the inverse and put the check at the top of the function 

```go
if feature.CreateGroup.IsEnabled() {
                respondWithAPIError(w, ctxServices.Log, errors.NewFeatureNotAvailable(""))
		return 
}
```

Comment (acosferreira) [reply pkg/routes/devicegroups.go:352]: I renamed the flag and move to edge-management scope, as we wont change anything on parity

File: unleash/features/feature.go (hunk 1 above)
Comment (acosferreira) [reply unleash/features/feature.go]: I renamed the flag and move to edge-management scope, as we wont change anything on parity

**Review by loadtheaccumulator — APPROVED** [review]
lgtm

**Review by ldjebran — APPROVED** [review]

**Pull request merged**